driver-core/src/main/com/mongodb/internal/connection/ByteBufBsonDocument.java

```diff
@@ -51,9 +52,11 @@
             documentsBuffer.position(documentsBuffer.position() - 4);
             ByteBuf documentBuffer = documentsBuffer.duplicate();
             documentBuffer.limit(documentBuffer.position() + documentSizeInBytes);
-            documents.add(new ByteBufBsonDocument(documentBuffer));
+            documents.add(new ByteBufBsonDocument(new ByteBufNIO(documentBuffer.asNIO())));
```

Comment:
This is a hack, but access to the documentBuffers contents are now made available to the Command Listeners without retaining those underlying buffers (which is the cause of netty reporting the leak).